### DISCUSSION TITLE
PNP Modern Search Text Web Part

### DISCUSSION THREAD

**Discussion by zeannait** [microsoft-search/pnp-modern-search#2877] [Q&A]
Using PNP Modern Search, how can I search in Text Web Part for content in selected site pages?

Thank you.

**Comment by kasperbolarsen** [comment]
I guess you should be able to use a query somewhat like this:
SPFxExtensionJson:{searchTerm} path:https://contoso.sharepoint.com/sites/PNP2877/SitePages/test2.aspx*

You can add additional pages to the query as required.
SPFxExtensionJson contains all the content of the page in a JSON format.

**Reply by zeannait** [reply]
Hi.  The link is giving me a "This site can’t be reached" message.  Please provide direction.  Thank you.

**Reply by kasperbolarsen** [reply]
You will have to replace the url with the url to the list or page you want to search.

**Reply by zeannait** [reply]
Got it!  Sorry new to SharePoint.  I made the change but the search result isn't looking into the Text Web Part content for the specified page.  Any other suggestion?

**Reply by kasperbolarsen** [reply]
Switch to the debug layout in order to see the content you get back from the API. 

If you get something back but you need more info then either add the SP Editor chrome extension to your browser or get the SharePoint search query tool and use one of the tools to inspect the results you get back.

**Reply by zeannait** [reply]
It's pointing to the correct page, but I'm not getting any results.  Is there a setting that I need to set for it to search for all content on a page?

**Reply by kasperbolarsen** [reply]
Try to see if you get any results if you use path:[insertsiteurl]* rather the the url to the page

**Reply by zeannait** [reply]
If I use {searchTerms}, I do get results, but for list and document libraries items.  I'm trying to get the results for pages that have the search term value inside the text web parts content.

**Reply by wobba** [reply]
Text in the default text web part for modern pages should be indexed and give results when searching on it.
